### PR TITLE
Add ray-on-golem stats command

### PR DESCRIPTION
**Pull request by lucekdudek**
- `ray-on-golem` now has `command` positional argument. Old behaviour is moved to `ray-on-golem webserver`
 ```bash
 $ ray-on-golem --help
usage: ray-on-golem [-h] {webserver,network-stats} ...

Ray on Golem's webserver.

positional arguments:
  {webserver,network-stats}

options:
  -h, --help            show this help message and exit
 ```
 ```bash 
$ ray-on-golem webserver --help
usage: ray-on-golem webserver [-h] [-p PORT] [--self-shutdown] [--registry-stats] [--no-registry-stats] [--no-self-shutdown]

options:
  -h, --help            show this help message and exit
  -p PORT, --port PORT  port for Ray on Golem's webserver to listen on, default: 4578
  --self-shutdown       flag to enable self-shutdown after last node termination, default: False
  --registry-stats      flag to enable collection of Golem Registry stats about resolved images, default: True
  --no-registry-stats
  --no-self-shutdown
  ```

 - new command `ray-on-golem network-stats`
 ```bash
$ ray-on-golem network-stats --help
usage: ray-on-golem network-stats [-h] [-d DURATION] [-log] CLUSTER_CONFIG_FILE

positional arguments:
  CLUSTER_CONFIG_FILE   Cluster config yaml

options:
  -h, --help            show this help message and exit
  -d DURATION, --duration DURATION
                        For how long in minutes to gather stats, default: 5
  -log, --enable-logging
                        flag to enable logging, default: False
```
```bash
$ ray-on-golem network-stats golem-cluster.dev.yaml
Gathering stats data for 5 minutes...
Gathering stats data done

Proposals count:
Initial: 81
Not blacklisted: 81
Passed Reject if max_average_usage_cost exceeds 1.5: 81
Passed Reject if price_initial exceeds 0.5: 81
Passed Reject if price_cpu_sec exceeds 0.0005: 81
Passed Reject if price_duration_sec exceeds 0.0005: 81
Scored: 81
Negotiated: 13

Negotiation errors:
Failed to send proposal response! Request timed out: 45
Failed to send proposal response! 500: {"message":"Failed to send response for Proposal [***]. Error: Countering Proposal [***] GSB error: GSB failure: Net: error forwarding message: Failed to find Node on relay server: Request timed out after 3000 ms.."}: 14
Failed to send proposal response! 500: {"message":"Failed to send response for Proposal [***]. Error: Countering Proposal [***] GSB error: GSB failure: Net: error forwarding message: Socket closed.."}: 5
Failed to send proposal response! 500: {"message":"Failed to send response for Proposal [***]. Error: Countering Proposal [***] GSB error: GSB failure: Net: error forwarding message: Connection timed out.."}: 3
Outbound rejected because: Everyone rule is disabled ; Audited-Payload rule requires manifest signature ; Partner rule requires node descriptor ; : 1
...
```

### CODE REVIEW

**Comment by mateuszsrebrny** [comment]
I would change `RUN_TIME` to `DURATION`, very similar, but RUNTIME is looking out of context here.
Also `stats` seems too general - I would change it to `network-stats`

I still struggle with the name webserver ;) but this is for different discussion, I would use something like service instead, and call the logs `ray-on-golem.log` & ray-on-golem-debug.log`

**Comment by mateuszsrebrny** [comment]
Oh, and I would change #1... #4 to names of properties from yaml that are taken into account there - so it is easier to translate into yaml tweaks

**Comment by lucekdudek** [comment]
> I would change `RUN_TIME` to `DURATION`, very similar, but RUNTIME is looking out of context here. Also `stats` seems too general - I would change it to `network-stats`
> 
> I still struggle with the name webserver ;) but this is for different discussion, I would use something like service instead, and call the logs `ray-on-golem.log` & ray-on-golem-debug.log`

> Oh, and I would change #1... #4 to names of properties from yaml that are taken into account there - so it is easier to translate into yaml tweaks

Done, see description @mateuszsrebrny

File: ray_on_golem/server/run.py [hunk 1]
```diff
@@ -128,29 +157,60 @@ async def ray_service_ctx(app: web.Application) -> None:
...
+        await stats_service.run(provider_config["parameters"], args.run_time)
+
+
 def main():
     prepare_tmp_dir()
     args = parse_sys_args()
 
-    logging.config.dictConfig(LOGGING_CONFIG)
+    if args.command == "webserver":
+        logging.config.dictConfig(LOGGING_CONFIG)
```
Comment (shadeofblue) [inline ray_on_golem/server/run.py:192]: I'd move these conditional pieces to separate `webserver` and `stats` functions for clarity's sake

**Comment by mateuszsrebrny** [comment]
maybe we could enhance `Gathering stats data...` with the duration?
`Gathering stats data for xx minutes...` :thinking: ?

**Review by approxit — APPROVED** [review]
Overall looks good, but updated `__init__`s would be a nice tweak.

Should we make `network_stats` service separate from `golem` service?

File: ray_on_golem/server/run.py [hunk 2]
```diff
@@ -20,30 +24,56 @@
 
 def parse_sys_args() -> argparse.Namespace:
     parser = argparse.ArgumentParser(description="Ray on Golem's webserver.")
-    parser.add_argument(
+    subparsers = parser.add_subparsers(dest="command")
```
Comment (shadeofblue) [inline ray_on_golem/server/run.py:27]: my feeling is that the name of the root parser should be more meaningful ... so either `commands` or `commands_parser` if we mean to be hyper correct...

```suggestion
    commands = parser.add_subparsers(dest="command")
```

File: ray_on_golem/server/run.py [hunk 3]
```diff
@@ -20,30 +24,56 @@
 
 def parse_sys_args() -> argparse.Namespace:
     parser = argparse.ArgumentParser(description="Ray on Golem's webserver.")
```
Comment (shadeofblue) [inline ray_on_golem/server/run.py:26]:
```suggestion
    parser = argparse.ArgumentParser(description="Ray on Golem")
```

File: ray_on_golem/server/main.py [hunk 4]
```diff
@@ -14,44 +14,50 @@
...
+@click.option(
+    "--self-shutdown",
+    is_flag=True,
+    show_default=True,
+    help="Enable self-shutdown after last node termination.",
+)
+@click.option(
+    "--registry-stats/--no-registry-stats",
+    show_default=True,
+    default=True,
```
[the remark this answers is not in the document]
Comment (approxit) [reply ray_on_golem/server/main.py:40]: In general this option is aimed only for developers, users don't need to play with this, so by design we want to send stats to registry. Note that users can always run yagna and webserver separately.

**Review by approxit — APPROVED** [review]

**Pull request merged**